pkg/containers/slice_queue.go

```diff
@@ -0,0 +1,109 @@
+package containers
```

Comment:
What's the motivation to implement such a FIFO queue? 
Do you have some performance tests?

1. Or can we just use channel ? 
2. Or implement the FIFO queue with the list + object pool?

Reply:
The motivation is:
- A Go channel is a bounded channel. Sometimes we need an unbounded channel, especially when the number of buffered elements can vary widely, or when it is sub-optimal for the producer to be blocked.